arangod/Replication/DatabaseTailingSyncer.cpp

```diff
@@ -81,21 +83,175 @@ Result DatabaseTailingSyncer::saveApplierState() {
   return rv;
 }
 
+Result DatabaseTailingSyncer::syncCollectionCatchup(std::string const& collectionName,
+                                                    TRI_voc_tick_t fromTick, double timeout,
+                                                    TRI_voc_tick_t& until, bool& didTimeout, 
+                                                    std::string const& context) {
+  TRI_ASSERT(ServerState::instance()->isDBServer());
+  TRI_ASSERT(!_unregisteredFromLeader);
+
+  try {
+    // always start from _initialTick
+    _initialTick = fromTick;
+    Result res = syncCollectionCatchupInternal(collectionName, timeout, false, until, didTimeout, context);
+    if (res.fail()) {
+      // if we failed, we can already unregister ourselves on the leader, so that
+      // we don't block WAL pruning
+      unregisterFromLeader();
```

Comment:
This call and the one below in line 106 make the corresponding calls in `SynchronizeShard.cpp` unnecessary. On the other hand, they also do not hurt, since we have `_unregisteredFromLeader`, so we might as well leave it as it is.

Reply:
Yes, it won't hurt because of `_unregisteredFromLeader`. I suggest to leave it as is as well.